common/src/main/include/control/PIDController.h

```diff
@@ -53,8 +56,11 @@ namespace control {
     double _integral;
     double _derivative;
     double _lastError;
-    double _threshIZone = -1;
+    double _avgError;
+    int _iterations = 0;    // Used to check if we have sufficient size in avgError.
 
+    double _threshIZone = -1;
+    double _threshAvg = 10; // Constant for now, might need to pass it in later
```

Comment:
This will have to be much lower, at the moment it's 10 metres. Should make it settable: `SetIsDoneThreshold` or something similar. Default it to a percentage of the setpoint (inside `SetSetpoint`, set the threshold to X% of the setpoint (maybe 5%?) )